prime-router/src/test/kotlin/common/HttpClientUtilsTests.kt

```diff
@@ -0,0 +1,110 @@
+package gov.cdc.prime.router.common
+
+import gov.cdc.prime.router.common.HttpClientUtils.Companion
+import io.ktor.client.HttpClient
+import junit.framework.TestCase.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.jupiter.api.BeforeEach
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+import kotlin.test.Test
+
+class HttpClientUtilsTests {
+
+    @BeforeEach
+    fun setup() {
+        val resetMethod: Method = Companion::class.java.getDeclaredMethod("reset")
+        resetMethod.trySetAccessible()
+        resetMethod.invoke(HttpClientUtils)
+    }
+
+    @Test
+    fun `test default values for clients and auth hash are null`() {
+        // httpClient should be initialized as null
+        val httpClientField: Field = HttpClientUtils::class.java.getDeclaredField("httpClient")
+        httpClientField.trySetAccessible()
+        assertEquals(null, httpClientField.get(HttpClient::class))
+
+        // httpClientWithAuth should be initialized as null
+        val httpClientWithAuth: Field = HttpClientUtils::class.java.getDeclaredField("httpClientWithAuth")
+        httpClientWithAuth.trySetAccessible()
+        assertEquals(null, httpClientWithAuth.get(HttpClient::class))
+
+        // accessTokenHash should be initialized as -1
+        val accessTokenHash: Field = HttpClientUtils::class.java.getDeclaredField("accessTokenHash")
+        accessTokenHash.trySetAccessible()
+        assertEquals(0, accessTokenHash.get(Int::class))
+    }
+
+    @Test
+    fun `test httpClient is a singleton and instantiating it does not impact value of other fields`() {
+        // create the httpClient singleton
+        val httpClient1 = HttpClientUtils.getDefaultHttpClient(null)
+        val httpClientObjRef1 = System.identityHashCode(httpClient1)
+
+        // request the httpClient obj again
+        val httpClient2 = HttpClientUtils.getDefaultHttpClient(null)
+        val httpClientObjRef2 = System.identityHashCode(httpClient2)
+
+        // should still be the same object
+        assertEquals(httpClientObjRef1, httpClientObjRef2)
+
+        // httpClientWithAuth should still be null
+        val httpClientWithAuth: Field = HttpClientUtils::class.java.getDeclaredField("httpClientWithAuth")
+        httpClientWithAuth.trySetAccessible()
+        assertEquals(null, httpClientWithAuth.get(HttpClient::class))
+
+        // accessTokenHash should still be -1
+        val accessTokenHash: Field = HttpClientUtils::class.java.getDeclaredField("accessTokenHash")
+        accessTokenHash.trySetAccessible()
+        assertEquals(0, accessTokenHash.get(Int::class))
```

Comment:
Comment says the accessTokenHash should be -1, but the assert says it should be 0

Reply:
oh! good catch. the hash for a null obj reference is always 0 and I didn't know that until I wrote this which is why it's not defaulting to 0. I'll take care of the comment.